edb/pgsql/compiler/dml.py

```diff
@@ -839,33 +838,81 @@ def compile_policy_check(
         dml_rvar = relctx.rvar_for_rel(dml_cte, ctx=ctx)
         relctx.include_rvar(ictx.rel, dml_rvar, path_id=subject_id, ctx=ictx)
 
-        cond_ref = clauses.compile_filter_clause(
-            policy_expr.expr, policy_expr.cardinality, ctx=ictx)
+        def raise_if(
```

Comment:
This looks solid and like it should work usefully.

I'd like to go a bit further in what we produce, though, if we can, and produce an error message that lists *all* of the failing deny rules. It will probably end up being a bit gnarlier than your approach, which is fairly elegant :P